### PR TITLE
add CMake support

### PR DESCRIPTION
**Pull request by tukss**
This adds support for building Rayleigh using CMake. It is purely optional and does not interfere with the traditional, `configure`-based approach.

You can try it like this:
```sh
cmake -B build
cmake --build build -j
```
The `rayleigh` executable will be placed in `build/src`.

Todo:

- [x] Fix `Run_Param_Header.F`. The Fortran flags are currently missing. The linker flags have a `SHELL:` prefix.
- [x] Document CMake build procedure including how to make `Debug` and `Release` builds.
- [x] Test on more systems, in particular MKL based ones.
- [x] Add CI tests for the CMake build.
- [x] Install the executable in `bin`.

### CODE REVIEW

**Comment by gassmoeller** [comment]
Oh, I am glad I didnt make the PR ;-). Did you see https://github.com/geodynamics/Rayleigh/pull/281? I am not sure what we can use from there, but it is worth checking. I am happy to test this branch on my system.

**Comment by tukss** [comment]
Thanks for pointing that out. I've checked and I think the new CMake is already more advanced. In particular the handling of `Run_Param_Headers.F` and MPI.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.28)
```
Comment (gassmoeller) [inline CMakeLists.txt:1]: is 3.28 necessary? I am running Ubuntu 22.04 and that ships 3.22 and I would guess there are plenty of older versions still floating around. In ASPECT we only require 3.13 to not exclude people with unmaintained institutional clusters. That is probably not necessary here, but 3.28 is probably quite uncommon.

**Comment by tukss** [comment]
As a bonus we get support for the new LLVM-based Intel `ifx` compiler for free.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+project(rayleigh LANGUAGES C Fortran)
```
Comment (gassmoeller) [inline CMakeLists.txt:5]: I would put the project command above the list command.

File: CMakeLists.txt (hunk 1 above)
Comment (gassmoeller) [inline CMakeLists.txt:1]: This file should get a license header as well.

**Review by gassmoeller — COMMENTED** [review]
The system already works well for me. Some small suggestions for the `CMakeLists`.

File: CMakeLists.txt (hunk 1 above)
Comment (tukss) [reply CMakeLists.txt]: I've just relaxed it to 3.23. I have to check what we need exactly.

Comment (tukss) [reply CMakeLists.txt]: 3.13 seems to work, too. I'll leave it there for now.

File: CMakeLists.txt (hunk 2 above)
Comment (tukss) [reply CMakeLists.txt:3]: Changed that.

**Comment by tukss** [comment]
It even works fine on Cray systems with the cray-fftw. The main thing I still want to try is pleiades.

File: CMakeLists.txt (hunk 1 above)
Comment (tukss) [reply CMakeLists.txt]: I've added license headers to both cmake files. Actually neither the `configure` nor the `Makefile` have a license header right now.

**Comment by gassmoeller** [comment]
I tested the build on the NSF/TACC systems Stampede3 and Frontera, both use Intel compilers and MKL. I used Intel icc 19.1 on Frontera, and Intel 2024 (the new clang based) on Stampede3. Both worked fine, except I needed to load the fftw3 module.

**Comment by tukss** [comment]
I think this is ready now.

When I added the test, I got an error with the macOS tester because it passed `-fallow-argument-mismatch` to the clang compiler there. The problem goes away by switching to openmpi, which supports the F08 interface. I don't recall why we used mpich before but we should talk about that.

**Comment by tukss** [comment]
@feathern We have a changelog entry now. Feel free to merge it at any time.

**Review by feathern — APPROVED** [review]
Looks good -- approved.

**Pull request merged**